src/core/server/elasticsearch/integration_tests/auditor.test.ts

```diff
@@ -0,0 +1,195 @@
+  let root: ReturnType<typeof createRootWithCorePlugins>;
+  let internalStart: InternalCoreStart;
+  const auditor = { add: jest.fn(), withAuditScope: jest.fn() };
+  const auditorFactory = jest.fn().mockReturnValue(auditor);
+
+  beforeAll(async () => {
+    servers = createTestServers({
+      adjustTimeout: (t) => jest.setTimeout(t),
+    });
+    esServer = await servers.startES();
+    root = createRootWithCorePlugins();
+    const internalSetup = await root.setup();
+
+    internalSetup.auditTrail.register({ asScoped: auditorFactory });
+    internalStart = await root.start();
+  });
+
+  afterAll(async () => {
+    await root.shutdown();
+    await esServer.stop();
+  });
+
+  afterEach(() => {
+    auditor.add.mockClear();
+    auditorFactory.mockClear();
+  });
+
+  describe('Legacy', () => {
+    describe('client', () => {
+      it('do not logs requests performed with un-scoped client', async () => {
+        await internalStart.elasticsearch.legacy.client.callAsInternalUser('ping');
+        expect(auditorFactory).toHaveBeenCalledTimes(0);
+        expect(auditor.add).toHaveBeenCalledTimes(0);
+      });
+
+      it('ScopedClusterClient.callAsInternalUser() writes into auditor', async () => {
+        const client = internalStart.elasticsearch.legacy.client.asScoped(
+          httpServerMock.createKibanaRequest()
+        );
+
+        await client.callAsInternalUser('ping');
+        expect(auditor.add).toHaveBeenCalledTimes(1);
+        expect(auditor.add).toHaveBeenCalledWith({
+          message: 'ping',
+          type: 'legacy.elasticsearch.call.internalUser',
+        });
+      });
+
+      it('ScopedClusterClient.callAsCurrentUser() writes into auditor', async () => {
+        const client = internalStart.elasticsearch.legacy.client.asScoped(
+          httpServerMock.createKibanaRequest()
+        );
+
+        await client.callAsCurrentUser('ping');
+        expect(auditor.add).toHaveBeenCalledTimes(1);
+        expect(auditor.add).toHaveBeenCalledWith({
+          message: 'ping',
+          type: 'legacy.elasticsearch.call.currentUser',
+        });
+      });
+    });
+
+    describe('custom client', () => {
+      it('do not logs requests performed with un-scoped client', async () => {
+        const customClient = internalStart.elasticsearch.legacy.createClient('custom');
+
+        await customClient.callAsInternalUser('ping');
+        expect(auditorFactory).toHaveBeenCalledTimes(0);
+        expect(auditor.add).toHaveBeenCalledTimes(0);
+      });
+
+      it('ScopedClusterClient.callAsInternalUser() writes into auditor', async () => {
+        const customClient = internalStart.elasticsearch.legacy.createClient('custom');
+        const client = customClient.asScoped(httpServerMock.createKibanaRequest());
+
+        await client.callAsInternalUser('ping');
+        expect(auditor.add).toHaveBeenCalledTimes(1);
+        expect(auditor.add).toHaveBeenCalledWith({
+          message: 'ping',
```

Comment:
Note the difference in audit log record format between the legacy & new clients:
- in the legacy client, it's not possible to access request method and path unless we auditor is integrated into `ElasticsearchClientLogging` class
- in the new client `RequestEvent` doesn't contain client method name

I'm inclining towards keeping this as is until we have the format requirements.